app/repositories/slick/implementations/SlickChatsRepository.scala

```diff
@@ -236,19 +239,43 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
         case (chatID, subject, fromAddress) => Some(CreateChatDTO(
           Some(chatID),
           Some(subject),
-          CreateEmailDTO(
+          UpsertEmailDTO(
             emailId = Some(emailId),
-            from = fromAddress,
+            from = Some(fromAddress),
             to = createEmailDTO.to,
             bcc = createEmailDTO.bcc,
             cc = createEmailDTO.cc,
             body = createEmailDTO.body,
-            date = Some(date))))
+            date = Some(date),
+            sent = Some(false))))
       }
       case None => None
     }
   }
 
+  def patchEmail(upsertEmailDTO: UpsertEmailDTO, chatId: String, emailId: String, userId: String): Future[Option[Email]] = {
```

Comment:
Please separate db.runs from DBIOActions by making a version of this method called "patchEmailAction" that returns a DBIOAction and another called "patchEmail" that simply calls db.run on said DBIOAction.